opm/parser/eclipse/RawDeck/StarToken.hpp

```diff
@@ -36,15 +36,39 @@ namespace Opm {
             return boost::lexical_cast<T>(valueString);
         }
         catch (boost::bad_lexical_cast&) {
-            throw std::invalid_argument("Unable to parse string" + valueString + " to typeid: " + typeid(T).name());
+            throw std::invalid_argument("Unable to convert string '" + valueString + "' to typeid: " + typeid(T).name());
         }
     }
 
     template <>
+    inline double readValueToken<double>(const std::string& valueString) {
+        // Eclipse supports Fortran syntax for specifying exponents of floating point
+        // numbers ('D' and 'E', e.g., 1.234d5) while C++ only supports the 'e' (e.g.,
+        // 1.234e5). the quick fix is to replace 'D' by 'E' and 'd' by 'e' before trying
+        // to convert the string into a floating point value. This may not be the most
+        // performant thing to do, but it is probably fast enough.
+        std::string vs(valueString);
```

Comment:
I'll measure how large the difference is for the Norne deck tomorrow. if the signal/noise ratio is insignificant, I'd prefer to keep is as proposed in the patch. (because it is less complex and hence less can go wrong.)